File: A/734-A.rs
fn main() {
    // 734-A - Anton and Danik
    use std::io;
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let mut a = 0;
    let mut d = 0;
    for c in input.trim().chars() {
        match c {
            'A' => a += 1,
            'D' => d += 1,
            _ => (),
        }
    }
    if a > d {
        println!("Anton");
    } else if a < d {
        println!("Danik");
    } else {
        println!("Friendship");
    }
}

File: cpp/A/734-A.cpp
#include <iostream>
using namespace std;

int main() {
  int n, c =0;
  string s;
  cin >> n >> s;
  for (int i = 0; i < n; i++) {
  if (s[i] == 'A') {
    c++;
  }else {
    c--;
  }
  if (c == 0) {
    cout << "Friendship" << endl;
  } else if (c > 0) {
    cout << "Anton" << endl;
  } else {
    cout << "Danik" << endl;
  }  }
}
